src/cherry_pipelines/evm/stablecoins/stablecoin_factory.py

```diff
@@ -0,0 +1,518 @@
+from cherry_etl.pipeline import run_pipeline
+from cherry_etl.writers.writer import create_writer
+from cherry_core import evm_signature_to_topic0, get_token_metadata_as_table
+from cherry_core.ingest import (
+    Query as IngestQuery,
+    QueryKind,
+)
+from cherry_core.ingest.evm import (
+    Query,
+    Fields,
+    BlockFields,
+    TransactionFields,
+    LogFields,
+    LogRequest,
+)
+
+from cherry_pipelines import db
+from cherry_pipelines.config import (
+    EvmConfig,
+    make_evm_table_name,
+)
+
+from clickhouse_connect.driver.asyncclient import AsyncClient
+
+EVENT_SIGNATURE = "Transfer(address indexed from, address indexed to, uint256 amount)"
+SCHEMA_VERSION = "1.0.0"
+PIPELINE_VERSION = "1.0.0"
+STABLECOIN_LIST = {
+    "circle_usdc_v1_ethereum": [
+        "Circle USDC v1",
+        "ethereum",
+        "circle_usdc_v1",
+        "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
+    ],
+    "curve_finance_crvusd_ethereum_v1_ethereum": [
+        "Curve Finance crvUSD v1",
+        "ethereum",
+        "curve_finance_crvusd_v1",
+        "0xf939E0A03FB07F59A73314E73794Be0E57ac1b4E",
+    ],
+    "ethena_usde_v1_ethereum": [
+        "Ethena USDe v1",
+        "ethereum",
+        "ethena_usde_v1",
+        "0x4c9EDD5852cd905f086C759E8383e09bff1E68B3",
+    ],
+    "first_digital_fdusd_v1_ethereum": [
+        "First Digital USD v1",
+        "ethereum",
+        "first_digital_fdusd_v1",
+        "0xc5f0f7b66764F6ec8C8Dff7BA683102295E16409",
+    ],
+    "frax_finance_frax_v1_ethereum": [
+        "Frax Finance Frax v1",
+        "ethereum",
+        "frax_finance_frax_v1",
+        "0x853d955aCEf822Db058eb8505911ED77F175b99e",
+    ],
+    "liquity_lusd_v1_ethereum": [
+        "Liquity LUSD",
+        "ethereum",
+        "liquity_lusd_v1",
+        "0x5f98805A4E8be255a32880FDeC7F6728C6568bA0",
+    ],
+    "maker_dai_v1_ethereum": [
+        "Maker DAI v1",
+        "ethereum",
+        "maker_dai_v1",
+        "0x6B175474E89094C44Da98b954EedeAC495271d0F",
+    ],
+    "paxos_usdp_v1_ethereum": [
+        "Paxos USDP v1",
+        "ethereum",
+        "paxos_usdp_v1",
+        "0x8E870D67F660D95d5be530380D0eC0bd388289E1",
+    ],
+    "paypal_pyusd_v1_ethereum": [
+        "Paypal pyUSD v1",
+        "ethereum",
+        "paypal_pyusd_v1",
+        "0x6c3ea9036406852006290770BEdFcAbA0e23A0e8",
+    ],
+    "decentralized_usdd_v1_ethereum": [
+        "Decentralized USDD v1",
+        "ethereum",
+        "decentralized_usdd_v1",
+        "0x3D7975EcCFc61a2102b08925CbBa0a4D4dBB6555",
+    ],
+    "trueusd_tusd_v1_ethereum": [
+        "TrueUSD TUSD v1",
+        "ethereum",
+        "trueusd_tusd_v1",
+        "0x0000000000085d4780B73119b644AE5ecd22b376",
+    ],
+    "world_liberty_financial_usd1_v1_ethereum": [
+        "World Liberty Financial USD1 v1",
+        "ethereum",
+        "world_liberty_financial_usd1_v1",
+        "0x8d0D000Ee44948FC98c9B98A4FA4921476f08B0d",
+    ],
+}
+
+
+async def create_protocol_table(
```

Comment:
Doesn't seem like this table is needed to be in the database, can just create a dataframe and pass it to the pipeline through custom step context and keep using it while the pipeline is running

Reply:
An even better way would be to collect this metadata using a script and harcode it here as a dict